velox/connectors/hive/HiveConnectorSplit.h

```diff
@@ -27,6 +27,10 @@ struct HiveConnectorSplit : public connector::ConnectorSplit {
   dwio::common::FileFormat fileFormat;
   const uint64_t start;
   const uint64_t length;
+
+  /// A map stores partition keys and values in string. Typed partition values
```

Comment:
perhaps,

```
Mapping from partition keys to values. Values are specified as strings formatted the same way as CAST(x as VARCHAR). Null values are specified as std::nullopt. Date values must be formatted using ISO 8601 as YYYY-MM-DD.
```

> All scalar types and date type are supported.

Is this so? Do we support real, double and timestamp? What's the format expected for these?

Reply:
Yes, real, double and timestamp are supported in Converter ([real and double](https://github.com/facebookincubator/velox/blob/main/velox/type/Conversions.h#L395-L397), [timestamp](https://github.com/facebookincubator/velox/blob/main/velox/type/Conversions.h#L575-L577)). The expected format should be the same as cast expects [real and double](https://github.com/facebookincubator/velox/blob/main/velox/expression/tests/CastExprTest.cpp#L341-L371), [timestamp](https://github.com/facebookincubator/velox/blob/main/velox/expression/tests/CastExprTest.cpp#L548-L556)

Reply:
I was wondering whether Spark support real, double, timestamp as partition keys.

Reply:
They are supported in Spark, but boolean type is not. There is one test on partition column types in Spark [link](https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetPartitionDiscoverySuite.scala#L703-L716).

Reply:
Got it. Would it be possible to add tests for all supported types in a follow-up ?

Reply:
Yes, I will follow up in a separate PR. Through first glance, misssing tests are on decimal and timestamp.

Reply:
@mbasmanova I double-checked and found only decimal was missing. Opened https://github.com/facebookincubator/velox/pull/8670 to add.